script/deploy/mainnet/BaseMainnetScript.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "forge-std/Script.sol";
+
+import {Addresses} from "contracts/utils/Addresses.sol";
+
+abstract contract BaseMainnetScript is Script {
+    uint256 public deployBlockNum = type(uint256).max;
+    bool isForked = false;
+
+    function run() external {
+        if (block.chainid != 1) {
+            revert("Not Mainnet");
+        }
+        // Will not execute script if after this block number
+        if (block.number > deployBlockNum) {
+            // console.log("Current block %s, script block %s", block.number, deployBlockNum);
+            return;
+        }
+
+        isForked = vm.envOr("IS_FORK", false);
```

Comment:
`vm.isContext(VmSafe.ForgeContext.ScriptDryRun)` might be a much cleaner way than environment variables for this one. Will make this change